cmake/setup_cached_variables.cmake

```diff
@@ -129,6 +129,17 @@ IF( NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Release" AND
 ENDIF()
 
 #
+# Force exactly one build type for multi-cofiguration generators.
+# Unfortunately CMake doesn't support custom configuration types so we just
+# call "DebugRelease" a "Release" build as well.
```

Comment:
Also, I am confused why you remove the different build types here. I thought this is necessary so that debug and release libs are built? Does this also influence the xcode generator?